File: rc-application/src/main/java/com/transkript/reportcard/business/service/ClassLevelService.java
package com.transkript.reportcard.business.service;


import com.transkript.reportcard.api.dto.ClassLevelDto;
import com.transkript.reportcard.api.dto.response.EntityResponse;
import com.transkript.reportcard.data.entity.ClassLevel;

import java.util.List;

public interface ClassLevelService {
    EntityResponse addClassLevel(ClassLevelDto classLevelDto);

    List<ClassLevelDto> getClassLevels();

    List<ClassLevelDto> getClassLevelsBySection(Long sectionId);

    ClassLevelDto getClassLevel(Long id);

    ClassLevel getClassLevelEntity(Long id);

    EntityResponse updateClassLevel(Long id, ClassLevelDto classLevelDto);

    void deleteClassLevel(Long id);
}

File: rc-application/src/main/java/com/transkript/reportcard/business/service/ClassLevelService.kt
package com.transkript.reportcard.business.service

import com.transkript.reportcard.api.dto.ClassLevelDto
import com.transkript.reportcard.api.dto.response.EntityResponse
import com.transkript.reportcard.data.entity.ClassLevel

interface ClassLevelService {
    fun addClassLevel(classLevelDto: ClassLevelDto?): EntityResponse?
    val classLevels: List<ClassLevelDto?>?
    fun getClassLevelsBySection(sectionId: Long?): List<ClassLevelDto?>?
    fun getClassLevel(id: Long?): ClassLevelDto?
    fun getClassLevelEntity(id: Long?): ClassLevel?
    fun updateClassLevel(id: Long?, classLevelDto: ClassLevelDto?): EntityResponse?
    fun deleteClassLevel(id: Long?)
}
